app/services/preserved_object_handler.rb

```diff
@@ -191,12 +195,31 @@ def update_online_version(validated=false, status=nil)
     handler_results.remove_db_updated_result unless transaction_ok
   end
 
-  def confirm_version_in_catalog
+  def confirm_online_version
     transaction_ok = with_active_record_transaction_and_rescue do
       po_db_object = PreservedObject.find_by!(druid: druid)
       pc_db_object = PreservedCopy.find_by!(preserved_object: po_db_object, endpoint: endpoint)
-      confirm_version_on_db_object(po_db_object, :current_version)
-      confirm_version_on_db_object(pc_db_object, :version)
+
+      if incoming_version > po_db_object.current_version && po_db_object.current_version == pc_db_object.version
+        handler_results.add_result(PreservedObjectHandlerResults::ARG_VERSION_GREATER_THAN_DB_OBJECT, po_db_object.class.name)
+        po_db_object.current_version = incoming_version
+        handler_results.add_result(PreservedObjectHandlerResults::ARG_VERSION_GREATER_THAN_DB_OBJECT, pc_db_object.class.name)
+        if moab_validation_errors.empty?
+          update_status(pc_db_object, 'ok')
+        else
+          update_status(pc_db_object, 'invalid')
+        end
+        update_preserved_copy_version_etc(pc_db_object, incoming_version, incoming_size, true)
+      elsif incoming_version == po_db_object.current_version && po_db_object.current_version == pc_db_object.version
+        handler_results.add_result(PreservedObjectHandlerResults::VERSION_MATCHES, po_db_object.class.name)
+        handler_results.add_result(PreservedObjectHandlerResults::VERSION_MATCHES, pc_db_object.class.name)
+      else
```

Comment:
this logic is flawed:  you could get here if po and pc versions don't match, also.    I'm not sure if we do the same thing in either case, but certainly you only output arg_version_less_than if it's true ...

Happy to pair on this;  it might be more efficient.